### ISSUE TITLE
Handling CoreWebView2OnPermissionRequested shows WebView2's prompt when using CoreWebView2PermissionState.Default

### ISSUE THREAD

**Issue by RendijsSmukulis** [MicrosoftEdge/WebView2Feedback#2406]
**Description**

I'm looking to re-implement the WebView2's Permissions dialog using our own controls. 

There are three user behaviors possible:

- Accept the permission request
- Deny the permission request
- Cancel out of the request (which will deny the current request, and prompt next time again)

Setting `CoreWebView2PermissionRequestedEventArgs.State` to `Allow` or `Deny` works as expected, and WebView2 prompt does not show up.

However, the third scenario (cancel the request) doesn't appear to be possible.

Either setting the state to `CoreWebView2PermissionState.Default` or not setting it all (only setting `CoreWebView2PermissionRequestedEventArgs.Handled` to true) leads to the WebView2 prompt showing up after ours is closed.

**Version**
SDK: 1.0.1222-prerelease
Runtime: evergreen
Framework: WPF
OS: Win10

**Repro Steps**

MainWindow.xaml:

```
<wpf:WebView2 Source="https://permission.site" Name="webview"></wpf:WebView2>
    </Grid>
</Window>
```

MainWindow.xaml.cs: 

```
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            this.InitializeComponent();
            this.webview.CoreWebView2InitializationCompleted += this.WebviewOnCoreWebView2InitializationCompleted;
        }

        private void WebviewOnCoreWebView2InitializationCompleted(object? sender, CoreWebView2InitializationCompletedEventArgs e)
        {
            this.webview.CoreWebView2.PermissionRequested += this.CoreWebView2OnPermissionRequested;
        }

        private void CoreWebView2OnPermissionRequested(object? sender, CoreWebView2PermissionRequestedEventArgs e)
        {
            Debug.WriteLine($"+++> Requested {e.PermissionKind} for uri {e.Uri}");
            var def = e.GetDeferral();
            e.State = CoreWebView2PermissionState.Default;
            e.Handled = true;
            def.Complete();
        }
    }
```

- Run the code
- Click the "Camera" button in the page, this will request the Camera permission

Expected: the handler sets the State to Default, permission is denied, no WebView2 prompt is shown

Actual: WebView2 prompt is shown.

[AB#39328742](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/39328742)

**Comment by champnic** [comment]
Thanks for the bug report @RendijsSmukulis - I've opened it on our backlog.

**Comment by vickiez** [comment]
Hi @RendijsSmukulis, can you use the experimental permission management APIs to accomplish what you need? https://github.com/MicrosoftEdge/WebView2Feedback/issues/2427#issuecomment-1399097515

**Comment by q71114** [comment]
Hi @vickiez, this seems to work well now with the [SavesInProfile](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2experimentalpermissionrequestedeventargs3?view=webview2-1.0.1619-prerelease) property on PermissionRequestedEventArgs.

**Comment by vickiez** [comment]
Glad to hear that @q71114!

**Comment by vickiez** [comment]
@q71114 @RendijsSmukulis The SavesInProfile property is now available in the latest release package, 1.0.1661.34. Thanks!
